Login Skeleton

AlexA0595 opened the pull request:
a basic login page. Additional add on as well

alexaa23 committed:
Login Skeleton

a basic login page.

veronicanashed commented on package.json:
```diff
@@ -3,19 +3,28 @@
     "@types/node": "^16.18.14",
     "@types/react": "^18.0.28",
     "@types/react-dom": "^18.0.11",
+    "@types/validator": "^13.7.13",
```
Are we using the dependency '@types/validator' anywhere in the code currently?

veronicanashed commented on package.json:
```diff
@@ -3,19 +3,28 @@
+    "formik": "^2.2.9",
+    "framer-motion": "^10.2.4",
     "react": "^18.2.0",
+    "react-bootstrap": "^2.7.2",
```
Are we using the dependency 'react-bootstrap' anywhere in the code currently?

veronicanashed commented on package.json:
```diff
@@ -3,19 +3,28 @@
+    "react-bootstrap": "^2.7.2",
     "react-dom": "^18.2.0",
     "react-scripts": "5.0.1",
+    "react-validation": "^3.0.7",
```
Are we using the dependency 'react-validation' anywhere in the code?

veronicanashed commented on package.json:
```diff
@@ -3,19 +3,28 @@
+    "react-validation": "^3.0.7",
     "typescript": "^4.9.5",
-    "web-vitals": "^2.1.4"
+    "validator": "^13.9.0",
```
Is validator being used?

veronicanashed commented on src/App.tsx:
```diff
@@ -1,11 +1,16 @@
+import Login from "./Login/Login";
 import {RoutesEnum} from "./Routes";
-import {HashRouter, Route, Routes} from "react-router-dom";
+import {HashRouter, Navigate, Route, Routes} from "react-router-dom";
```
We don't need the 'Navigate' import

veronicanashed commented on src/App.tsx:
```diff
@@ -20,6 +25,7 @@ function App() {
                             </PageContainer>
                         }
                     />
+                    <Route path={RoutesEnum.LOGIN} element={<Login setAuth={setAuth}/>} />
```
Ok I see this approach - this would be fine if we were doing a really basic auth, but since we're dealing with tokens of some kind, we really should use context and not prop drill here.
There's an example of context in NavigationContext, but essentially what we want is an AuthContext, where the **state** will include the user details (Could be a USER, which has a username, name, email and contactNumber) and the single use token we receive from backend.
The reason why this is important is that backend will require us to send through the authentication any time we hit an endpoint, which could be in any component nested in the <App /> component. If this is the case, we want to just pull the token from context, and not have to prop drill it through every single layer

veronicanashed commented on src/App.tsx:
```diff
@@ -1,11 +1,16 @@
+import {HashRouter, Navigate, Route, Routes} from "react-router-dom";
 import PageContainer from "./Components/PageContainer/PageContainer";
 function App() {
+    const [auth, setAuth] = useState(false);
```
Ideally we don't want to create state here and pass it down to every level

veronicanashed commented on src/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+            ease: easing,
+        },
+    },
+};
```
This looks awesome - we should use this for the sign up wizard too *** (putting this comment here for me to remember)

veronicanashed commented on src/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+                        <Typography variant="body2" sx={{ color: "text.secondary" }}>
+                            OR
+                        </Typography>
+                    </Divider>
```
Can we keep this in, but just comment it out until we decide on implementing the google sign in (since it looks like tutor wants us to demo, and we don't want to show him something not fully implemented yet)

veronicanashed commented on src/Login/LoginForm.tsx:
```diff
@@ -0,0 +1,170 @@
+    },
+};
+
+// @ts-ignore
```
when we remove the prop-drill, we can also get rid of this //@ts-ignore, and passing the state setter

veronicanashed commented on src/Login/LoginForm.tsx:
```diff
@@ -0,0 +1,170 @@
+        onSubmit: () => {
+            console.log("submitting...");
+            setTimeout(() => {
+                console.log("submited!!");
```
we can remove console.log's

veronicanashed commented on src/Contexts/NavigationContext.tsx:
```diff
@@ -33,6 +33,11 @@ export const NavigationContextContextProvider = ({children}: any) => {
             text: 'Test',
             route: RoutesEnum.TEST
         },
+        {
+            icon: <LogoutIcon />,
+            text: 'Login',
+            route: RoutesEnum.LOGIN
+        },
```
NavigationContext is what we see in the side nav bar, so this is adding a side nav icon to the login page, with the text 'Login': 
<img width="233" alt="image" src="https://user-images.githubusercontent.com/126940746/224670413-9ad6ba2c-1c3a-4807-97c8-79d6c43e1095.png">

What you could do is use the 'Logout (Inactive)' text (and remove the inactive), and just set the route to RoutesEnum.LOGIN. 
I.e. we click LOGOUT -> it routes the user to the login page again
if you want to get really fancy (but you don't have to at this stage), you could add a confirmation box when they click this

veronicanashed commented on src/Login/LoginForm.tsx:
```diff
@@ -0,0 +1,170 @@
+import React, { useState } from "react";
```
Could we pop the Login folder under Pages?
Pages -> Login -> {Login}, {LoginForm}

veronicanashed commented on src/Login/LoginForm.tsx:
```diff
@@ -0,0 +1,170 @@
+                                label="Remember me"
+                            />
+
+                            <Link
+                                component={RouterLink}
+                                variant="subtitle2"
+                                to="#"
+                                underline="hover"
+                            >
+                                Forgot password?
+                            </Link>
```
Let's perhaps comment this out too, until we can sort out the backend stuff

alexaa23 committed:
Updated

Made the changes required

alexaa23 committed:
Updated

alexaa23 committed:
Auth0

This is the best I can do with Auth0. I had a lot of trouble because the files needed are in js not jsx it didn't like that

AlexA0595 committed:
Updates

Update on the login branch

AlexA0595 committed:
Check Password

A verify link page doesn't send link yet if wanted and password reset page as well to access the password reset page click the send link button then it will take you to the reset page. we can you use either one
